modules/nf-core/propr/propd/templates/propd.R

```diff
@@ -63,6 +63,98 @@ read_delim_flexible <- function(file, header = TRUE, row.names = 1, check.names
     return(mat)
 }
 
+#' @return output Data frame
+read_delim_flexible_raw <- function(file, header = TRUE, row.names = NULL, check.names = TRUE){
+  
+  ext <- tolower(tail(strsplit(basename(file), split = "\\\\.")[[1]], 1))
+  
+  if (ext == "tsv" || ext == "txt") {
+    separator <- "\\t"
+  } else if (ext == "csv") {
+    separator <- ","
+  } else {
+    stop(paste("Unknown separator for", ext))
+  }
+  
+  mat <- read.delim(
+    file,
+    sep = separator,
+    header = header,
+    row.names = row.names,
+    check.names = check.names
+  )
+  
+  return(mat)
```

Comment:
Hey, actually it might be great if the input matrix for this module is already formatted so that the first column has the gene ids, and then the rest of columns are just the samples.
In this way you can convert read_delim_flexible_raw into read_delim_flexible, and remove the original read_delim_flexible.
If you do so, you may have to do a bit of groovy inside the test `when` section, so that the second column of the test data is removed before running the module.

Reply:
I can see your point, since read_delim_flexible_raw is highly similar to read_delim_flexible. But I don't think the benefit outweighs complicating the module code. Another approach would be conditional logic in the read_delim_flexible function to switch between the two behaviors. But I will not make this update.